ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+{
+    memset((char *)(&bd_xp->pField[i_camp].reserved_2) +
+               MM_OFFSET_RESERVAT2_OFFSET_NOM_ESTES,
+           0, 4);
+}
+
+static void
+MM_InitializeBytesExtendedFieldNameFields(struct MM_DATA_BASE_XP *bd_xp,
+                                          MM_EXT_DBF_N_FIELDS i_camp)
+{
+    memset((char *)(&bd_xp->pField[i_camp].reserved_2) +
+               MM_OFFSET_RESERVED2_EXTENDED_NAME_SIZE,
+           0, 1);
+}
+
+static short int MM_return_common_valid_DBF_field_name_string(char *cadena)
+{
+    char *p;
+    short int error_retornat = 0;
+
+    if (!cadena)
+        return 0;
+    //strupr(cadena);
+    for (p = cadena; *p; p++)
+    {
+        (*p) = (char)toupper(*p);
+        if ((*p >= 'A' && *p <= 'Z') || (*p >= '0' && *p <= '9') || *p == '_')
+            ;
+        else
+        {
+            *p = '_';
+            error_retornat |= MM_FIELD_NAME_CHARACTER_INVALID;
+        }
+    }
+    if (cadena[0] == '_')
+    {
+        cadena[0] = '0';
+        error_retornat |= MM_FIELD_NAME_FIRST_CHARACTER_;
+    }
+    return error_retornat;
+}
+
+static short int MM_ReturnValidClassicDBFFieldName(char *cadena)
+{
+    size_t long_nom_camp;
+    short int error_retornat = 0;
+
+    long_nom_camp = strlen(cadena);
+    if ((long_nom_camp < 1) ||
+        (long_nom_camp >= MM_MAX_LON_CLASSICAL_FIELD_NAME_DBF))
+    {
+        cadena[MM_MAX_LON_FIELD_NAME_DBF - 1] = '\0';
+        error_retornat |= MM_FIELD_NAME_TOO_LONG;
+    }
+    error_retornat |= MM_return_common_valid_DBF_field_name_string(cadena);
+    return error_retornat;
+}
+
+static MM_BOOLEAN
+MM_CheckClassicFieldNameEqual(const struct MM_DATA_BASE_XP *data_base_XP,
+                              const char *classical_name)
+{
+    MM_EXT_DBF_N_FIELDS i;
+
+    for (i = 0; i < data_base_XP->nFields; i++)
+    {
+        if ((strcasecmp(data_base_XP->pField[i].ClassicalDBFFieldName,
+                        classical_name)) == 0 ||
+            (strcasecmp(data_base_XP->pField[i].FieldName, classical_name)) ==
+                0)
+            return TRUE;
+    }
+    return FALSE;
+}
+
+static char *MM_GiveNewStringWithCharacterInFront(const char *text,
+                                                  char caracter)
+{
+    char *ptr;
+    size_t i;
+
+    if (!text)
+        return nullptr;
+
+    i = strlen(text);
+    if ((ptr = calloc_function(i + 2)) == nullptr)
+        return nullptr;
+
+    *ptr = caracter;
+    memcpy(ptr + 1, text, i + 1);
+    return ptr;
+}
+
+static char *MM_SetSubIndexFieldNam(char *nom_camp, MM_EXT_DBF_N_FIELDS index,
+                                    size_t ampladamax)
+{
+    char *NomCamp_SubIndex;
+    char *_subindex;
+    char subindex[15];
+    size_t sizet_subindex;
+    size_t sizet_nomcamp;
+
+    NomCamp_SubIndex = calloc_function(ampladamax * sizeof(char));
+    if (!NomCamp_SubIndex)
+        return nullptr;
+
+    strcpy(NomCamp_SubIndex, nom_camp);
+
+    sprintf(subindex, sprintf_UINT64, (GUInt64)index);
```

Comment:
General remark. We should only use snprintf, not sprintf. The later is much too dangerous regarding the possibility of buffer overflows:
```suggestion
    snprintf(subindex, sizeof(subindex), sprintf_UINT64, (GUInt64)index);
```

Reply:
I adopt it everywhere. In our old compiler (we use Visual Studio 2005 and trying to migrate) this function doesn't exist. But it's easy to create a similar one with same name.

Reply:
aouch 2005... You might be able to use ``_snprintf`` there, although old _snprintf() wasn't C99 compliant. See https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/snprintf-snprintf-snprintf-l-snwprintf-snwprintf-l?view=msvc-140#remarks

Reply:
I mean use _snprintf only in non GDAL context. We definitely want to use the C99 compliant function when possible